library/Fission/Web/IPFS/CID.hs

```diff
@@ -9,13 +9,12 @@ import Servant
 
 import Fission.Prelude
 import Fission.Models
+import Fission.User.CID.Query as UserCIDQuery
 
 type API = Get '[JSON, PlainText] [CID]
 
 allForUser :: MonadDB m => Entity User -> ServerT API m
 allForUser (Entity userId _) = runDB do
-  hashes <- select <| from \userCid -> do
-    where_ (userCid ^. UserCidUserFk ==. val userId)
-    return (userCid ^. UserCidCid)
-
-  return (unValue <$> hashes)
+  userCids <- UserCIDQuery.getByUserId userId
+  let cids = getInner userCidCid <$> userCids
+  return cids
```

Comment:
These are the same:

```haskell
let cids = getInner userCidCid <$> userCids
return cids
```

```haskell
getInner userCidCid <$> userCids
```

They'll also probably have the same runtime performance (because rewrite rules will optimize it), but you don't actually need to set that variable. We know that it's the return of the function by it just being the last line.

Reply:
Oh, sorry, I should clarify something about `do` notation.

```haskell
do
  let foo = bar
  -- more stuff

do
  foo <- return bar
  -- more stuff

return bar >>= \foo -> -- more stuff
```

I wish they had called it something like `const` to disambiguate it from `let ... in`, but here we are

Reply:
So looking at this. I dont think your first comment is correct?
Changing this:
```haskell
let cids = getInner userCidCid <$> userCids
return cids
```

To this:
```haskell
getInner userCidCid <$> userCids
```
resulted in a compilation error.

However this compiled just fine
```haskell
  return (getInner userCidCid <$> userCids)
```

Is that right? or did I miss something?

Reply:
Oh, yeah, indeed you're right 😅 the `let` creates a `return` but you don't have one otherwise 🤦‍♀ Juuust ignore me